src/main/world/rgb-palette.shader

```diff
@@ -19,13 +19,13 @@ void fragment() {
 	vec4 rgba_in = texture(TEXTURE, UV);
 	
 	// color defaults to the black replacement color
-	vec3 rgb_out = black.rgb;
+	vec4 rgb_out = black;
 	
 	// mix in other colors based on the red, green and blue components of the source image
-	rgb_out = mix(rgb_out, red.rgb, rgba_in.r);
-	rgb_out = mix(rgb_out, green.rgb, rgba_in.g);
-	rgb_out = mix(rgb_out, blue.rgb, rgba_in.b);
+	rgb_out = mix(rgb_out, red.rgba, rgba_in.r);
+	rgb_out = mix(rgb_out, green.rgba, rgba_in.g);
+	rgb_out = mix(rgb_out, blue.rgba, rgba_in.b);
 	
 	// Assign final color for the pixel, and preserve transparency
-	COLOR = vec4(rgb_out.rgb, rgba_in.a);
+	COLOR = vec4(rgb_out.rgb, rgba_in.a * rgb_out.a);
```

Comment:
This change seems valuable (was it to allow alpha components in the rgb-palette shader?) but make sure to document it in your commit comment.

Reply:
Added to commit comment.